bigml/ensemble.py

```diff
@@ -253,6 +283,108 @@ def list_models(self):
         """
         return self.model_ids
 
+    def predict_probability(self, input_data, by_name=True,
+                            voting_style=PROBABILITY_CODE,
+                            missing_strategy=LAST_PREDICTION):
+
+        """For classification problems, Predicts a probabilistic "score" for
+        each possible output class, based on input values.  The input
+        fields must be a dictionary keyed by field name.  For
+        classifications, the output is a list with one floating point
+        element for each possible class, ordered in sorted class-name
+        ordering.
+
+        For regressions, the output is a single element vector
+        containing the prediction.
+
+        :param input_data: Input data to be predicted
+        :param by_name: Boolean that is set to True if field_names (as
+                        alternative to field ids) are used in the
+                        input_data dict
+        :param voting_style: numeric key code indicating how the scores
+                             should be produced:
+              0 - majority vote - The sum of models predicting a given class
+                  divided by the total number of models
+                  PLURALITY_CODE
+              1 - Scores estimated from the class confidence at the leaves
+                  of each of the constituient models.
+                  CONFIDENCE_CODE
+              2 - Average Lapalace-smoothed probabilitiy of all models'
+                  leaf node distributions:
+                  PROBABILITY_CODE
+        :param missing_strategy: LAST_PREDICTION|PROPORTIONAL missing strategy for
+                                 missing fields
+
+        """
+        if self.regression:
+            output = [self.predict(input_data,
+                                   by_name=by_name,
+                                   method=voting_style,
+                                   missing_strategy=missing_strategy)]
+        elif self.boosting is not None:
+            probabilities = self.predict(input_data,
+                                         by_name=by_name,
+                                         missing_strategy=missing_strategy,
+                                         add_probability=True)['probabilities']
+
+            probabilities.sort(key=lambda x: x['prediction'])
+            output = [p['probability'] for p in probabilities]
```

Comment:
> better change p into probability

I can do this, but just FYI it seems the linter does not complain about one-char names in list/dict comprehensions, presumably because their meaning is sufficiently obvious.  If it's against convention here, though, I will make a pass to change them all.